ocaml/xapi/xapi_session.ml

```diff
@@ -661,6 +665,49 @@ let login_no_password_common ~__context ~uname ~originator ~host ~pool
   ignore (Client.Pool.get_all ~rpc ~session_id) ;
   session_id
 
+let login_no_password_common ~__context ~uname ~originator ~host ~pool
+    ~is_local_superuser ~subject ~auth_user_sid ~auth_user_name
+    ~rbac_permissions ~db_ref ~client_certificate =
+  Context.with_tracing ~originator ~__context __FUNCTION__ @@ fun __context ->
+  let internal_xapi_master_to_xapi_slave_login =
+    (xapi_internal_originator, true, true, None)
+  in
+  let is_valid_session session_id =
+    try
+      (* Call an API function to check the session is still valid *)
```

Comment:
Could we use a cheaper call if we are not using the result? There is `Pool.hello` for example.

Reply:
Yeah I'd imagine so. I used Pool.get_all as that is what is used in the original login_no_password_common (https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/xapi_session.ml#L658-L661) and I presumed that was using the cheapest call as it is also ignoring the result. I presume this can also be changed then, there it is only used to force the time to update?

Reply:
Looking at Pool.hello now, would it actually be cheaper? It requires a host_uuid and host_address so wouldn't we need to do another call to get that? We have the host ref as a parameter but nothing else.